kod/object/active/holder/nomoveon/battler/player/user.kod

```diff
@@ -2414,7 +2414,15 @@ messages:
       }
 
       AddPacket(4,iFlags);
-      AddPacket(4,0); % Item rarity grade
+
+      if IsClass(what,&Item)
```

Comment:
A little less fragile in the face of future changes would be this:
```
    rarity = 0;
    if IsClass(what,&Item) {
      rarity = Send(what,@GetRarity);
    }
    AddPacket(4, rarity);
```

i.e. there is only one path that adds bytes, so there's nothing to get out of sync

Reply:
Modified to be less fragile as suggested.